packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -224,6 +227,11 @@ class RouteSettings {
   /// If null, the route is anonymous.
   final String name;
 
+  /// The arguments map that this route was created with.
+  ///
+  /// If null, there were no arguments given for this route.
```

Comment:
This should have a few paragraphs detailing the motivation for this feature, e.g. talking about how if the app is handling a URL, the path part of the URL will become the name of the route, and the query arguments will end up as a `Map<String, List<String>>` map in the arguments, talking about why it's a list (query arguments aren't guaranteed to be unique), about whether the arguments are URL-decoded (presumably they are), etc.